src/metabase/query_processor/middleware/annotate.clj

```diff
@@ -490,7 +495,9 @@
    [:maybe [:sequential Col]]
    [:fn
     {:error/message ":cols with unique names"}
-    #(su/empty-or-distinct? (map :name %))]])
+    (fn [cols]
+      (or (empty? cols)
+          (apply distinct? (map :name cols))))]])
```

Comment:
This deserves a name since it's been used 5-6 times.

Reply:
I added a new util function `u/empty-or-distinct?`